packages/node-terra/src/utils/terra-helper.ts

```diff
@@ -61,43 +108,102 @@ export async function fetchTerraBlocksArray(
 }
 
 export async function getTxInfobyHashes(
-  api: TerraClient,
+  api: LCDClient,
   txHashes: string[],
 ): Promise<TxInfo[]> {
   return Promise.all(
     txHashes.map(async (hash) => {
-      return api.txInfo(hash);
+      return api.tx.txInfo(hashToHex(hash));
     }),
   );
 }
 
-export async function getEventsByTypeFromBlock(
-  api: TerraClient,
+export async function getTerraEvents(
+  api: LCDClient,
+  blockInfo: BlockInfo,
+): Promise<TerraEvent[]> {
+  const txHashes = blockInfo.block.data.txs;
+  if (txHashes.length === 0) {
+    return [];
+  }
+  const txInfos = await getTxInfobyHashes(api, txHashes);
+  const eventInfos: TerraEvent[] = [];
+  for (const ti of txInfos) {
+    const txLogs = ti.logs;
+    for (const log of txLogs) {
+      const eventInfo: TerraEvent = {
+        event: log.eventsByType,
+        blockNumber: +blockInfo.block.header.height,
+        blockHash: blockInfo.block_id.hash,
+        blockTimestamp: ti.timestamp,
+        msgIndex: log.msg_index,
+        txInfo: ti,
+        log: log,
+        transactionHash: ti.txhash,
+      };
+      eventInfos.push(eventInfo);
+    }
+  }
+  return eventInfos;
+}
+
+export async function getContractExecutionTxInfos(
+  api: LCDClient,
   blockInfo: BlockInfo,
-): Promise<EventsByType[]> {
+): Promise<TerraCall[]> {
   const txHashes = blockInfo.block.data.txs;
   if (txHashes.length === 0) {
     return [];
   }
   const txInfos = await getTxInfobyHashes(api, txHashes);
-  const txLogs = txInfos.map((txInfo) => txInfo.logs);
-  const txLogsFlat = ([] as TxLog[]).concat(...txLogs);
-  const events = txLogsFlat.map((txLog) => txLog.eventsByType);
-  return events;
+  const callInfos: TerraCall[] = [];
+  for (const txInfo of txInfos) {
+    for (const msg of txInfo.tx.body.messages) {
+      if (msg.toData()['@type'] === '/terra.wasm.v1beta1.MsgExecuteContract') {
+        const data = msg.toData() as MsgExecuteContract.Data;
```

Comment:
Should we be using https://github.com/terra-money/terra.js/blob/9e5f553/src/core/Msg.ts#L238 here.